src/js/components/RuleLabel.react.js

```diff
@@ -0,0 +1,42 @@
+import type { Props } from '../containers/AppContainer.react';
+import { RulePropertyUtils } from '../utils/RulePropertyUtils';
+
+/**
+ * Copyright 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+const React = require('react');
+const level_1 = '•';
+const level_2 = '✘';
+const level_3 = '✔';
+
+class RuleLabel extends React.Component<Props, State> {
+  render() {
+    return (
+      <div>
+        {this.props.className && this.props.htmlFor
+          ? '<label className="' +
+            this.props.className +
```

Comment:
In this case `this.props` is of type `Props`, which in this context is simply the `Props` type of the `AppContainer`. It does not have a `className` property. 

You need to extend the container `Props` by importing it with a new name:

```js
import type { Props as BaseProps } from '../containers/AppContainer.react';
type Props = BaseProps & { className: string ... 
```

Please check the example on `PropertyPicker.rect.js`.

Reply:
So I should add ```type Props = BaseProps & { className: string, htmlFor: string};``` ?